ipam/allocator_test.go

```diff
@@ -590,6 +590,45 @@ func TestGetSameAddress(t *testing.T) {
 	}
 }
 
+func TestPoolAllocationReuse(t *testing.T) {
+	for _, store := range []bool{false, true} {
+		a, err := getAllocator(store)
+		assert.NoError(t, err)
+
+		// First get all pools until they are exhausted to
+		pList := []string{}
+		pool, _, _, err := a.RequestPool(localAddressSpace, "", "", nil, false)
```

Comment:
I feel RequestPool call can be simplified like below
pool, _, _, err := a.RequestPool(localAddressSpace, "", "", nil, false)====> remove this line 
		for err == nil {====> this will be true first time anyways.
			pool, _, _, err = a.RequestPool(localAddressSpace, "", "", nil, false)
                        pList = append(pList, pool)
		}

Reply:
Ah, but I don't want to append the value pool (which may be an empty string) to the list when err != nil. That would be wrong. So I'd need a test for err inside the body of the loop. But that's 2 extra lines to mirror the loop test. :(